core/workspace_svg.js

```diff
@@ -944,6 +944,7 @@ Blockly.WorkspaceSvg.prototype.loadAudio_ = function(filenames, name) {
     if (ext && audioTest.canPlayType('audio/' + ext[1])) {
       // Found an audio format we can play.
       sound = new window['Audio'](filename);
+      sound.setAttribute('preload', 'auto');
```

Comment:
Out of curiosity, why you are not using the "required" method (Web Audio API) from Apple to load the sound files on iOS to make sure that they are working as expected ?

See:
https://developer.apple.com/library/safari/documentation/AudioVideo/Conceptual/Using_HTML5_Audio_Video/Device-SpecificConsiderations/Device-SpecificConsiderations.html
https://developer.apple.com/library/safari/documentation/AudioVideo/Conceptual/Using_HTML5_Audio_Video/PlayingandSynthesizingSounds/PlayingandSynthesizingSounds.html#//apple_ref/doc/uid/TP40009523-CH6-SW1

The current solution is "unsupported" for desktop browsers and will throw the given error message.
I'm not a friend of browser specific solutions, but in this case it seems there is no other way.